crates/iceberg/src/io/storage_s3.rs

```diff
@@ -152,11 +151,7 @@ pub(crate) fn s3_config_parse(mut m: HashMap<String, String>) -> Result<S3Config
 }
 
 /// Build new opendal operator from give path.
-pub(crate) fn s3_config_build(
-    client: &reqwest::Client,
-    cfg: &S3Config,
-    path: &str,
```

Comment:
Please keep the `crates/iceberg/src/io/storage_s3.rs` unchanged, since the client inside the S3 is used to optimize the performance.